src/v2/api/group/GroupService.ts

```diff
@@ -69,4 +78,40 @@ export class GroupService {
 
     return results;
   }
+
+  async addUnregistered(groupId: string, body: EmailDTO) {
+    await this.userRepository.addUnregistered(groupId, body);
+  }
+
+  async verifyStudent(groupId: string, email: string, data: ApproveDTO){
+    const user = await this.userRepository.getByEmail(email);
+
+    if (user.student.groupId !== groupId) {
+      throw new NoPermissionException();
+    }
+
+    await this.studentRepository.updateState(user, data);
```

Comment:
the name of method is wrong. you need to create abstract method update, that can update every possible field